drivers/i2s/i2s_ll_stm32.c

```diff
@@ -0,0 +1,1170 @@
+	if (ret < 0) {
+		return ret;
+	}
+	k_sem_give(&stream->sem);
+
+	/* Assure cache coherency before DMA read operation */
+	DCACHE_CLEAN(stream->mem_block, mem_block_size);
+
+	if (stream->master) {
+		LL_I2S_SetTransferMode(cfg->i2s, LL_I2S_MODE_MASTER_TX);
+	} else {
+		LL_I2S_SetTransferMode(cfg->i2s, LL_I2S_MODE_SLAVE_TX);
+	}
+
+	/* remember active TX DMA channel (used in callback) */
+	active_dma_tx_channel[stream->dma_channel] = dev;
+
+	ret = start_dma(dev_data->dev_dma, stream->dma_channel,
+			&stream->dma_cfg,
+			stream->mem_block,
+			(void *)LL_SPI_DMA_GetRegAddr(cfg->i2s),
+			stream->cfg.block_size);
+	if (ret < 0) {
+		SYS_LOG_ERR("Failed to start TX DMA transfer: %d", ret);
+		return ret;
+	}
+
+	LL_I2S_EnableDMAReq_TX(cfg->i2s);
+
+	LL_I2S_EnableIT_ERR(cfg->i2s);
+	LL_I2S_Enable(cfg->i2s);
+
+	return 0;
+}
+
+static void rx_stream_disable(struct stream *stream, struct device *dev)
+{
+	const struct i2s_stm32_cfg *cfg = dev->config->config_info;
+	struct i2s_stm32_data *const dev_data = DEV_DATA(dev);
+	struct device *dev_dma = dev_data->dev_dma;
+
+	LL_I2S_Disable(cfg->i2s);
+	LL_I2S_DisableDMAReq_RX(cfg->i2s);
+	LL_I2S_DisableIT_ERR(cfg->i2s);
+
+	dma_stop(dev_dma, stream->dma_channel);
+	if (stream->mem_block != NULL) {
+		k_mem_slab_free(stream->cfg.mem_slab, &stream->mem_block);
+		stream->mem_block = NULL;
+	}
+
+	active_dma_rx_channel[stream->dma_channel] = NULL;
+}
+
+static void tx_stream_disable(struct stream *stream, struct device *dev)
+{
+	const struct i2s_stm32_cfg *cfg = dev->config->config_info;
+	struct i2s_stm32_data *const dev_data = DEV_DATA(dev);
+	struct device *dev_dma = dev_data->dev_dma;
+
+	LL_I2S_Disable(cfg->i2s);
+	LL_I2S_DisableDMAReq_TX(cfg->i2s);
+	LL_I2S_DisableIT_ERR(cfg->i2s);
+
+	dma_stop(dev_dma, stream->dma_channel);
+	if (stream->mem_block != NULL) {
+		k_mem_slab_free(stream->cfg.mem_slab, &stream->mem_block);
+		stream->mem_block = NULL;
+	}
+
+	active_dma_tx_channel[stream->dma_channel] = NULL;
+}
+
+static void rx_queue_drop(struct stream *stream)
+{
+	size_t size;
+	void *mem_block;
+
+	while (queue_get(&stream->mem_block_queue, &mem_block, &size) == 0) {
+		k_mem_slab_free(stream->cfg.mem_slab, &mem_block);
+	}
+
+	k_sem_reset(&stream->sem);
+}
+
+static void tx_queue_drop(struct stream *stream)
+{
+	size_t size;
+	void *mem_block;
+	unsigned int n = 0;
+
+	while (queue_get(&stream->mem_block_queue, &mem_block, &size) == 0) {
+		k_mem_slab_free(stream->cfg.mem_slab, &mem_block);
+		n++;
+	}
+
+	for (; n > 0; n--) {
+		k_sem_give(&stream->sem);
+	}
+}
+
+#ifdef CONFIG_I2S_1
+static struct device DEVICE_NAME_GET(i2s_stm32_1);
+
+static void i2s_stm32_irq_config_func_1(struct device *dev);
+
+static const struct i2s_stm32_cfg i2s_stm32_config_1 = {
+	.i2s = (SPI_TypeDef *) CONFIG_SPI_1_BASE_ADDRESS,
+	.pclken = {
+		.enr = LL_APB2_GRP1_PERIPH_SPI1,
+		.bus = STM32_CLOCK_BUS_APB2,
+	},
+#ifdef CONFIG_I2S_STM32_USE_PLLI2S_ENABLE
+	.i2s_clk_sel = LL_RCC_I2S2_CLKSOURCE_PLLI2S,
```

Comment:
LL_RCC_I2S1_CLKSOURCE_PLLI2S?

Reply:
No, this one is correct. But thanks to your observation I was able to find a couple of mistakes in others. So, basically:
 SPI/I2S 1/4/5 are on APB2 and must use LL_RCC_I2S2_CLKSOURCE_PLLI2S
 SPI/I2S 2/3 are on APB1 and must use LL_RCC_I2S1_CLKSOURCE_PLLI2S

I noticed that in my code SPI/I2S 2/3 are wrong. So, thanks!